src/Network/Network/Models/AzureFirewallPolicy/PSAzureFirewallPolicy.cs

```diff
@@ -42,5 +47,77 @@ public class PSAzureFirewallPolicy : PSTopLevelResource
 
         public PSAzureFirewallPolicySku Sku { get; set; }
+
+        public PSAzureFirewallPolicySNAT Snat { get; set; }
+
+        private const string IANAPrivateRanges = "IANAPrivateRanges";
+
+        public string[] PrivateRange
+        {
+            get
+            {
+                return Snat?.PrivateRanges.ToArray();
+            }
+            set
+            {
+                if (value != null)
+                    ValidatePrivateRange(value);
+                Snat = new PSAzureFirewallPolicySNAT() { PrivateRanges = value };
+            }
+        }
+
+        [JsonIgnore]
+        public string PrivateRangeText
+        {
+            get { return JsonConvert.SerializeObject(PrivateRange, Formatting.Indented); }
+        }
+
+        #region Private Range Validation
+        private void ValidatePrivateRange(string[] privateRange)
+        {
+            foreach (var ip in privateRange)
+            {
+                if (ip.Equals(IANAPrivateRanges, StringComparison.OrdinalIgnoreCase))
+                    continue;
+
+                if (ip.Contains("/"))
+                    ValidateMaskedIpAddress(ip);
+                else
+                    ValidateSingleIpAddress(ip);
+            }
+        }
+
+        private void ValidateSingleIpAddress(string ipAddress)
+        {
+            IPAddress ipVal;
+            if (!IPAddress.TryParse(ipAddress, out ipVal) || ipVal.AddressFamily != System.Net.Sockets.AddressFamily.InterNetwork)
+            {
+                throw new AzPSArgumentException(String.Format("\'{0}\' is not a valid private range ip address", ipAddress), "IpAddress", ErrorKind.UserError);
+            }
+        }
+
+        private void ValidateMaskedIpAddress(string ipAddress)
+        {
+            var split = ipAddress.Split('/');
+            if (split.Length != 2)
+                throw new AzPSArgumentException(String.Format("\'{0}\' is not a valid private range ip address", ipAddress), "IpAddress", ErrorKind.UserError);
+
+            // validate the ip
+            ValidateSingleIpAddress(split[0]);
+
+            // validate mask
+            var bit = 0;
+            if (!Int32.TryParse(split[1], out bit) || bit < 0 || bit > 32)
+                throw new AzPSArgumentException(string.Format("\'{0}\' is not a valid private range ip address, subnet mask should between 0 and 32", ipAddress), "IpAddress", ErrorKind.UserError);
+
+            // validated that unmasked bits are 0
+            var splittedIp = split[0].Split('.');
+            var ip = Int32.Parse(splittedIp[0]) << 24;
+            ip = ip + Int32.Parse(splittedIp[1]) << 16 + Int32.Parse(splittedIp[2]) << 8 + Int32.Parse(splittedIp[3]);
+            if (ip << bit != 0)
+                throw new AzPSArgumentException(String.Format("\'{0}\' is not a valid private range ip address, bits not covered by subnet mask should be all 0", ipAddress), "IpAddress", ErrorKind.UserError);
```

Comment:
It's encouraged that put these exception messages into azure-powershell/src/Network/Network/Properties/Resources.resx

Reply:
Updated accordingly